### DISCUSSION TITLE
[terraform] code completion not working

### DISCUSSION THREAD

**Discussion by milosgajdos** [zed-industries/zed#18692] [Feature Requests]
I've installed the latest version of Terraform extension, but for some reason code completions just don't work for me in Zed at all.

I have the following config:
```
  "lsp": {
    "terraform-ls": {
      "initialization_options": {
        "experimentalFeatures": {
          "prefillRequiredFields": true
        }
      }
    }
  }
```

I have a similar setup in vim with the same version of the terraform lsp and things work like a charm there. I can't seem to find anything related to this in Zed log, unfortunately. Could be a silent error or maybe my config is invalid 🤷‍♂️ the LSP does recognised the syntax though which I suppose is _something_.

I have the code completion setting enabled 

```
  "show_completions_on_input": true,
```

**Comment by omarjatoi** [comment]
Are you seeing anything in `debug: open language server logs`?

**Reply by milosgajdos** [reply]
There is a lot of noise from other language servers, but `terraform-ls` seems quiet.

I am wondering if that extension is even installed and where
```
ls -lh ~/Library/Application\ Support/Zed/extensions/work/terraform
total 0
```

If the binary is not there how come I'm seeing terraform syntax highlighted correctly? Zed pretends the extension is installed:

<img width="795" alt="Screenshot 2024-10-03 at 18 19 14" src="https://github.com/user-attachments/assets/656745c3-c2d1-4f6e-9e19-5b20e7d8b486">

Very strange and probably very broken, too

**Reply by omarjatoi** [reply]
I just installed the terraform extension and cloned a random repository (https://github.com/Azure/terraform).

Before opening the project I saw `.../Zed/extensions/work/terraform` empty as well, but after opening a file Zed started downloading the terraform language server automatically, and I was able to confirm:
```
ls -lh ~/Library/Application\ Support/Zed/extensions/work/terraform/
total 0
drwxr-xr-x@ 4 anon  staff   128B Oct  3 13:24 terraform-ls-v0.34.3
```

After that, autocomplete seems to be working for me. My recommendation would be to try and uninstall the extension, re-install, and then open your project and open a `.tf` file and see if things work?

**Reply by milosgajdos** [reply]
Oof, ok, so after I had restarted Zed (by quitting it) I can now see the LSP binary has been downloaded and the LSP server pretends to be started
```
2024-10-03T18:25:33.882014+01:00 [INFO] attempting to start language server "terraform-ls", path: "/Users/milosgajdos/foobar-tf", id: 2
2024-10-03T18:25:38.845525+01:00 [INFO] starting language server process. binary path: "/Users/milosgajdos/Library/Application Support/Zed/extensions/work/terraform/terraform-ls-v0.34.3/terraform-ls", working directory: "/Users/milosgajdos/code/foobar-tf", args: ["serve"]
```

Autocomplete now works like a charm. Ok, I didn't know I had to quit the editor when I install an extension. TIL.

**Reply by omarjatoi** [reply]
FWIW, I did not add anything to my Zed configuration

**Reply by omarjatoi** [reply]
Awesome! Glad you got it working 🙂

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
